book/src/developers/designs/identifiable_secrets.md

```diff
@@ -0,0 +1,34 @@
+# Identifiable Secrets
+
+Kanidm tokens should have a unique pattern, making them easy to recognize. This is crucial for security systems that aim to prevent incorrect credential storage. Without a distinct pattern, like with bare JWTs that look like any base64-encoded data, we risk false alarms.
+
+## The Kanidm pattern
+
+> **RFC Discussion here**
+>
+> Another option could be to use the pattern `kanidm_<CREDENTIAL>`. This way, users can figure out what kind of token they have on their own.
+
+```text
+kanidm_<TYPE>_<CREDENTIAL>
```

Comment:
What benefit does the type serve here do you think? Does it make it easier for people to identify what the token belonged to?  Couldn't we just have a "reflector" command line tool that shows what's in the variety of tokens to help people ID what token it was? 

Otherwise  Ithink most of this work is needed in compact_jwt, because I think being able to generically have a way to prefix tokens there would be best for reuse in other applications too

Reply:
Yeah, I'm easy either way - if we can easily do the reflector thing and cover all the various tokens we generate - that'd work too 😄

Reply:
Well the reflector is more so an admin can look at a token and say "what is it". 

So I think kanidm_<credential> would be sufficient for us? What do you think?

Reply:
Seems legit to me!